lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelResult.cxx

```diff
@@ -204,7 +243,8 @@ Scalar LinearModelResult::getRSquared() const
   const Scalar RSS = residuals.computeRawMoment(2)[0];
   // get outputSample and SYY
   // See https://stats.stackexchange.com/questions/26176/removal-of-statistically-significant-intercept-term-increases-r2-in-linear-mo
-  // In case there is no intercept convention for R^2 is to have the ration between sum of squared predicted over sum of squared real
+  // In case there is no intercept convention for R^2 is to have the ratio 
+  // between sum of squared predicted over sum of squared real
```

Comment:
I don't understand this comment.

Reply:
This comment explains the "if" statement at:

https://github.com/openturns/openturns/blob/aeddc45aab96a92bc5210e7f3cb10551f7fe230c/lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelResult.cxx#L211

According to [stats.stackexchange.com](https://stats.stackexchange.com/questions/26176/removal-of-statistically-significant-intercept-term-increases-r2-in-linear-mo), there are two definitions of the R2 coefficient.

- If there is an intercept, then the R2 measures the way the predictions of the linear model improve over the output mean. Moreover, the sample mean of the predictions is equal to the sample mean of the output observations in this case. This is why the denominator is equal to the sample mean _centered_ squared residuals, which is evaluated thanks to `computeCentralMoment()`:

https://github.com/openturns/openturns/blob/aeddc45aab96a92bc5210e7f3cb10551f7fe230c/lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelResult.cxx#L214

- Otherwise, it would make no sense to compare to the mean. Hence, the convention is to use the sample mean _raw_ squared residuals, which is evaluated thanks to `computeRawMoment()`:

https://github.com/openturns/openturns/blob/aeddc45aab96a92bc5210e7f3cb10551f7fe230c/lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelResult.cxx#L212

I was not aware of this detail: thanks for pointing it!

Reply:
Ok, got it, thanks!